src/apps/common/hiredis_libevent2.c

```diff
@@ -49,7 +49,7 @@ struct redisLibeventEvents {
   struct event *rev, *wev;
   int rev_set, wev_set;
   char *ip;
-  int port;
+  uint16_t port;
```

Comment:
I think you're going to need to revert this change (revert the whole file).

While an ideal world would have this port be represented as.... a port, which in TCP or UDP protocol networking is always unsigned 16 bits, this variable specifically is used in a third party API (Redis's redisAsyncConnect) which is defined as

```
redisAsyncContext *redisAsyncConnect(const char *ip, int port) {
    redisOptions options = {0};
    REDIS_OPTIONS_SET_TCP(&options, ip, port);
    return redisAsyncConnectWithOptions(&options);
}
```
with `REDIS_OPTIONS_SET_TCP` defined as
```
#define REDIS_OPTIONS_SET_TCP(opts, ip_, port_) do { \
        (opts)->type = REDIS_CONN_TCP;               \
        (opts)->endpoint.tcp.ip = ip_;               \
        (opts)->endpoint.tcp.port = port_;           \
    } while(0)
```
and `redisOptions`  defined as

```

typedef struct {
    /*
     * the type of connection to use. This also indicates which
     * `endpoint` member field to use
     */
    int type;
    /* bit field of REDIS_OPT_xxx */
    int options;
    /* timeout value for connect operation. If NULL, no timeout is used */
    const struct timeval *connect_timeout;
    /* timeout value for commands. If NULL, no timeout is used.  This can be
     * updated at runtime with redisSetTimeout/redisAsyncSetTimeout. */
    const struct timeval *command_timeout;
    union {
        /** use this field for tcp/ip connections */
        struct {
            const char *source_addr;
            const char *ip;
            int port;
        } tcp;
        /** use this field for unix domain sockets */
        const char *unix_socket;
        /**
         * use this field to have hiredis operate an already-open
         * file descriptor */
        redisFD fd;
    } endpoint;

    /* Optional user defined data/destructor */
    void *privdata;
    void (*free_privdata)(void *);

    /* A user defined PUSH message callback */
    redisPushFn *push_cb;
    redisAsyncPushFn *async_push_cb;
} redisOptions;
```


It's *trivially obviously* wrong for this field to use `int`, but that's just how c-language codebases do things in a lot of situations. The extreme amount of manual work needed to do basic type-system style work in c-language is what I would personally point to as the underlying cause for this abuse of the `int` type in this situation, but there's no getting around it.

So for this file, probably better to just leave it as `int` to work with how Redis does it.

Perhaps we can make an improvement here some other time.